server/example_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"gocloud.dev/health"
+	"gocloud.dev/requestlog"
+	"gocloud.dev/server"
+)
+
+func ExampleServer_New() {
+	// This example is used in https://gocloud.dev/howto/server/
+
+	// Use the constructor function to create the server.
+	srv := server.New(http.DefaultServeMux, nil)
+
+	// Register a route.
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, "Hello, World!")
+	})
+
+	// Start the server.
+	srv.ListenAndServe(":8080")
+}
+
+func ExampleServer_RequestLogger() {
+	// This example is used in https://gocloud.dev/howto/server/
+
+	// Create a logger, and assign it to the RequestLogger field of a
+	// server.Options struct.
+	srvOptions := &server.Options{
+		RequestLogger: requestlog.NewNCSALogger(os.Stdout, func(error) {}),
+	}
+
+	// Pass the options to the Server constructor.
+	srv := server.New(http.DefaultServeMux, srvOptions)
```

Comment:
Probably should add `defer srv.Shutdown()` here and elsewhere, right?

Reply:
🙃

Reply:
I added the defer calls to `Shutdown`, not sure if they're in the right place. I stuck it right after creating the server.

Reply:
Sorry, missed this on first pass:

`defer srv.Shutdown()` is probably not right since `ListenAndServe` blocks forever. The more correct code would be:

```go
srv := server.New(http.DefaultServeMux, srvOptions)
go func() {
	interrupt := make(chan os.Signal, 1)
	signal.Notify(interrupt, os.Interrupt)
	for { // in a loop, because the interrupt could be sent before ListenAndServe starts
		<-interrupt
		srv.Shutdown()
	}
}()
```

But stepping back, most "cloud-native"™️ deployments will likely behind a load balancer that will handle the graceful shutdown of requests. I think the hard interrupt is fine, so I think we can leave `Shutdown` out of these examples.

Because this is subtle, I recommend we add my above snippet to the `ExampleServer_Shutdown` godoc function, but not discuss it in the how-to guide, since it doesn't seem like an 80% use case. WDYT?

Reply:
SGTM. Sorry for my first misleading comment.